toolchain/check/impl.h

```diff
@@ -0,0 +1,18 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_IMPL_H_
+#define CARBON_TOOLCHAIN_CHECK_IMPL_H_
+
+#include "toolchain/check/context.h"
+#include "toolchain/sem_ir/ids.h"
+
+namespace Carbon::Check {
+
+// Builds and returns a witness table for the impl `impl_id`.
```

Comment:
FWIW, right now you return an InterfaceWitness, which I assume would contain the table, but I'd tend to say that it's returning a witness (or very precisely "InterfaceWitness") rather than a witness table. When it says "returns a witness table", I'd expect just that. Not sure if this is something that'll change, but might be worth thinking about how much precision to use in the comments.

Reply:
Eventually this will be a different kind of witness, once we support the more general  `impl Type as Constraint`, not merely `impl Type as Interface`, but it'll always be a witness, so I'll just drop the "table".

Reply:
```suggestion
// Builds and returns a witness for the impl `impl_id`.
```